addons/membership/membership.py

```diff
@@ -136,6 +137,11 @@ def _state(self, cr, uid, ids, name, args, context=None):
             res[line.id] = state
         return res
 
+    def _set_state(self, cr, uid, ids, field_name, field_value, args, context=None):
+        cr.execute('UPDATE membership_membership_line '\
+                       'SET state=%s '\
+                       'WHERE id IN %s', (field_value, tuple([ids]),))
+        return True
```

Comment:
hello,
The purpose to add 'fnct_inv' is that when `invoice_refund` function will call the state of account invoice is not changed so, that time to change the state of membership we have to add funtion fnct_inv and it will update state field's value.
if we want to remove fnct_inv, for that we have to make state field store=True and we have to update it's value every time when invoice state is change.